middleware/kubernetes/controller.go

```diff
@@ -371,6 +374,7 @@ func (dns *dnsControl) PodIndex(ip string) []interface{} {
 func (dns *dnsControl) EndpointsList() api.EndpointsList {
 	epl, err := dns.epLister.List()
 	if err != nil {
+		log.Printf("[ERROR] Could not get endpoint list: %s", err)
```

Comment:
Can we make these methods just return the `error` and potentially print from the caller?